### ISSUE TITLE
Install fails on brand new Raspberry Pi due to logname failing

### ISSUE THREAD

**Issue by merehap** [jochemin/raspnode#5]
Running "sudo ./raspnode.sh" fails on my system with the error message "logname: no login name", the same message that the logname command fails with. I worked around this with replacing logname in the script with $USER. Perhaps the script should use something other than logname, or have a fallback?

This is with the latest Raspberry Pi 3, Model B from CanaKit.

**Comment by jochemin** [comment]
If I change logname with $USER it returns me root as the script is launched with sudo. 

logname works fine in my Raspberry with raspbian, it would be interesting to know if this problem is general or only occurs to you.

![image](https://user-images.githubusercontent.com/6348955/37698832-ed59f090-2ce4-11e8-938e-8b80243581a9.png)

**Comment by merehap** [comment]
Yeah, you're right it was returning root (that did allow things to proceed, but not really in the right way).

I think the problem is general to anyone using the same version of Raspberry Pi from the same company. That said, it is clearly a bug with the specific Raspberry Pi distro rather than your script, so maybe you'd still want to just close this issue if there is no clear workaround.

**Comment by jochemin** [comment]
Ok, so your distro was not raspbian?

**Comment by merehap** [comment]
No, it is Raspbian. I'm just not sure why logname would be failing on a vanilla, unmodified install. When I said "distro" I just meant any modifications that CanaKit or others would have made to base Raspian that would cause logname to fail.

**Comment by jochemin** [comment]
In CanaKit web says they send an 8Gb microsd with NOOBS:
![image](https://user-images.githubusercontent.com/6348955/37756856-42856ff6-2dab-11e8-827f-0e575cf1e18c.png)

**Comment by merehap** [comment]
NOOBS just lets you choose which OSes you want to install. I only selected Raspbian from their menu.

**Comment by jochemin** [comment]
Ok, I would like to know if this problem is with Raspbian installed from NOOBS and occurs always. 

If someone has run the script in a Raspbian installed from NOOBS please, comment.

**Comment by Amiga500** [comment]
I have the same issue with Raspbian downloaded on official site https://www.raspberrypi.org/downloads/raspbian/ . I almost sure that is the same installed from NOOBS. I used a workaround and installed all like a charm ;)

**Comment by merehap** [comment]
I just created a bug on Raspbian's bug tracker for this: https://bugs.launchpad.net/raspbian/+bug/1758480

**Comment by michiel1972** [comment]
I had the same issue on my Raspberry 3b. Within the script, I replaced $(logname) with $LOGNAME and that works.
